components/livesplit/liveSplitManager.ts

```diff
@@ -378,19 +379,23 @@ export class LiveSplitManager {
     }
 
     private _addMissionTime(time: Seconds) {
+        let computedTime = time
         // always add at least minimum
         if (time <= this._resetMinimum) {
+            computedTime = this._resetMinimum
             this._currentMissionTotalTime += this._resetMinimum
             this._campaignTotalTime += this._resetMinimum
         } else if (time > 0 && time <= 1) {
             // if in game time is between 0 and 1, add full second
+            computedTime = 1
             this._currentMissionTotalTime += 1
             this._campaignTotalTime += 1
         } else {
             // important to always floor before adding time
             this._currentMissionTotalTime += Math.floor(time)
             this._campaignTotalTime += Math.floor(time)
         }
+        return computedTime
     }
```

Comment:
Added return type and simplified the logic a bit. Also, fixed the fact that the returned time wasn't floored
```ts
    private _addMissionTime(time: Seconds): Seconds {
        let computedTime = Math.floor(time)
        
        // always add at least minimum, which is usually 0 except on cutscenes where
        // you can gain an advantage by restarting in cs (bangkok, sgail specific starts)
        if (time <= this._resetMinimum) {
            computedTime = this._resetMinimum
        } else if (time > 0 && time <= 1) {
            // if in game time is between 0 and 1, add full second
            computedTime = 1
        }

        this._currentMissionTotalTime += computedTime
        this._campaignTotalTime += computedTime
        return computedTime
    }
```